sdk/core/azure-core/azure/core/pipeline/transport/_base.py

```diff
@@ -409,12 +413,22 @@ def prepare_multipart_body(self):
         main_message.add_header("Content-Type", "multipart/mixed")
         if boundary:
             main_message.set_boundary(boundary)
-        for i, req in enumerate(requests):
+
+        for req in requests:
             part_message = Message()
-            part_message.add_header("Content-Type", "application/http")
-            part_message.add_header("Content-Transfer-Encoding", "binary")
-            part_message.add_header("Content-ID", str(i))
-            part_message.set_payload(req.serialize())
+            if req.multipart_mixed_info:
+                content_index = req.prepare_multipart_body(content_index=content_index)
+                part_message.add_header("Content-Type", req.headers['Content-Type'])
+                payload = req.serialize()
+                # We need to remove the ~HTTP/1.1 prefix along with the added content-length
+                payload = payload[payload.index(b'--'):]
+            else:
+                part_message.add_header("Content-Type", "application/http")
+                part_message.add_header("Content-Transfer-Encoding", "binary")
+                part_message.add_header("Content-ID", str(content_index))
```

Comment:
It looks to me content_index only changes in non-multipart behavior?

Reply:
This function is recursive (see line 420) - so `content_index` is only applied on the "leaf" parts.